### DISCUSSION TITLE
Option do perform controller shift

### DISCUSSION THREAD

**Discussion by fpytloun** [zwave-js/zwave-js-ui#3028] [General]
I am about to migrate from Fibaro HCL. Existing Z-Wave network has tons of devices, sometimes placed on hardly reachable places.

Only way how to perform the migration would be to perform exclusion on already unreliable network and do inclusion again with new controller. For most devices this will require diassembly of light switches, etc. and most likely falling back to hard-reset of devices where exclusion will not work. Overall it's work for many hours, therefore I am trying to find some other way.

Z-Wave supports secondary controllers and promoting to primary controller, somewhere also collect controller shift. But I didn't find any mention of zwave-js support for this feature. Is it difficult to implement?

My original controller is Fibaro HCL, it cannot do local backups (only cloud) but in theory.. if I have full backup of controller  locally (including security keys, etc.), would it be possible to use that to transfer that configuration into zwave-js (RaZberry 7) and taking over that original Fibaro controller?

**Comment by robertsLando** [comment]
AFAIK this is not possible yet but @AlCalzone can give you more details

**Comment by cornellrwilliams** [comment]
I just use the Z-Wave PC Controller software to include devices as secondary controllers and to perform controller shifts. This software is made by Silicon Labs which is the company that owns the Z-Wave protocol. According to the documentation this software is an **_example of the functionality a Z-Wave-enabled PC application should have_**, so it supports all of the latest Z-Wave features. Plus its free and runs on Windows, Linux, and MacOS. Because of how easy it is to setup and run the software I just use it to include devices as secondary controllers then I connect them to Z-Wave JS UI and copy over my security keys. Once inside of Z-Wave JS UI I can then control my devices as I would in any other configuration. Installing the software is the longest part of the process. Once installed though it takes less than 60 secs to perform a shift or just include a device as a secondary controller. 

Also I did notice that the software will not allow you to migrate the suc/sis role to any node other than 1. These roles responsible for keeping up with the issued node IDs and the routing table. If these go offline you won't be able to heal your network or include and exclude devces. I am not sure if this is a software limitation in the program but I believe in the older IQ panels you could manually assign these roles to different nodes.

Here's what it looks like with both of my controllers included into the same network. I have tested this functionality on other hubs but didn't have any available to show at this time so I just use (2) USB Sticks.
![Secondary Controller](https://user-images.githubusercontent.com/92619869/228576523-dc69de5a-b5cf-4be9-a234-1a78eb3a0b16.PNG)

**Reply by fpytloun** [reply]
Thank you, this looks promising.

So from what I read, I would need to setup Z/IP Gateway on my Raspberry Pi with RaZberry HAT. Then I could connect to Z/IP via PC Controller but unfortunately this is only Windows app (so maybe run Simplicity Studio which can install this software inside Wine on my Mac). Then I should be able to include RaZberry as a secondary controller via Fibaro HCL and make controller shift 🤔 
It doesn't seem to be that simple but probably simpler than running around house with screwdriver in hand.
Fortunately there's still Homeassistant so if this fails, I can still perform migration by exclusing/including slowly (room by room) while not breaking up much things (like house heating).

**Reply by cornellrwilliams** [reply]
You need to download simplicity studio for windows, Mac, or linux. Then once you install Simplicity Studio you run it then add the Z-Wave PC Controller tool. Here's a guide on how to set it up. https://community.silabs.com/s/article/How-to-get-Z-Wave-PC-Controller-and-Zniffer-tools?language=en_US


You would then launch the program and connect to your Razberry. If the fibaro has a shift mode you would put it in that mode. Then inside the PC Controller Software click learn then click nwi. Your Razberry will join the network the network and the primary controller role will be shifted to the new device. If you don't have a shift option you would just have to put the controller in inclusion mode.

**Comment by cornellrwilliams** [comment]
So I was able to successfully migrate from one hub to Z-WavejsUI.  The best part is that I didn't have to exclude and reinclude any of the devices. Also because this process doesn't require you do anything special on your hub it should work for all hubs. The only caveat is that it does require you use an extra piece of software. Installing the software imo is the longest part of the process. 

I made a step by step video explaining what I did.
https://youtu.be/71w0OLO6zxY
